Source/EasyNetQ/Internals/TaskHelpers.cs

```diff
@@ -96,54 +125,61 @@ public static class TaskHelpers
                 {
                     tcs.SetException(exception);
                 }
+
                 return tcs.Task;
             };
         }
 
-        public static Task FromCancelled()
+        /// <summary>
+        ///     This is an internal API that supports the EasyNetQ infrastructure and not subject to
+        ///     the same compatibility as public APIs. It may be changed or removed without notice in
+        ///     any release. You should only use it directly in your code with extreme caution and knowing that
+        ///     doing so can result in application failures when updating to a new EasyNetQ release.
+        /// </summary>
+        public static void AttachCancellation<T>(
+            this TaskCompletionSource<T> taskCompletionSource, CancellationToken cancellationToken
+        )
         {
-            var tcs = new TaskCompletionSource<object>();
-            tcs.SetCanceled();
-            return tcs.Task;
-        }
+            if (!taskCompletionSource.Task.CreationOptions.HasFlag(TaskCreationOptions.RunContinuationsAsynchronously))
```

Comment:
What is the reason for this check? At first glance, the method doesn’t care and works for any options.

Reply:
I've double-checked and yes it is fine.